src/runtime_src/core/pcie/driver/linux/CMakeLists.txt

```diff
@@ -1,11 +1,24 @@
 # SPDX-License-Identifier: Apache-2.0
-# Copyright (C) 2019-2021 Xilinx, Inc. All rights reserved.
+# Copyright (C) 2019-2022 Xilinx, Inc. All rights reserved.
 #
 add_subdirectory(include)
 
 set(XRT_KERNEL_HEADERS
   include/xocl_ioctl.h
 )
 
+set(QDMA_LIB_DESTINATION_DIR
+  ${CMAKE_CURRENT_SOURCE_DIR}/xocl/lib/libqdma/
+)
+
+function(CopyQDMALIB)
+ message(STATUS "Copying QDMA lib file from $ENV{QDMA_LIB_DIR}")
+ file(COPY $ENV{QDMA_LIB_DIR} DESTINATION ${QDMA_LIB_DESTINATION_DIR})
+endfunction()
+
+if(DEFINED ENV{QDMA_LIB_DIR})
+ CopyQDMALIB()
+endif()
```

Comment:
Shouldn't this be implemented as a target dependency?   Now copying will be a CMake runtime, but it should be  a dependency based copying and what ever target requires libqdma should trigger the copying. 

Also please implement the env as a CMake variable and add it to cmake_flags in build.sh.